### PR TITLE
Domains: New domain status page design

### PR DESCRIPTION
**Pull request by hambai**
This is a very rough implementation of the new domain status page design. It's feature flagged behind `domains/new-status-design` and currently doesn't implement all the possible statuses but is a good start. the idea is to merge this so we can split-up further work into smaller PRs

#### Changes proposed in this Pull Request

* Implement the basic domain status component
* feature flag all of it

#### Testing instructions

* enable `domains/new-status-design` (or run calypso locally) and check that the wpcom domain is showing the new domain status screen
* check that newly registered domains show the "Activating..." status in the first 30 minutes
* check that unverified contact info notice is shown for unverified contacts
* check that expired notice (and "Action required") is shown for expired domains
* some icons might not show properly if https://github.com/Automattic/wp-calypso/pull/39438 is not deployed by then

see: p99Zz8-UK-p2
depends on https://github.com/Automattic/wp-calypso/pull/39438 (a bunch of material icons)

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/domain-types/wpcom-domain-type.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+				external={ ! isWpcomDomain }
+				onClick={
+					isWpcomDomain ? this.handleChangeSiteAddressClick : this.handleEditSiteAddressClick
+				}
+			>
+				<MaterialIcon icon="language" className="wpcom-domain-type__multiline-nav-item-icon" />
+				<div>
+					<div>
+						{ isWpcomDomain
+							? this.props.translate( 'Change WordPress.com Site Address' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/wpcom-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **26** times:
`translate( 'WordPress.com Blog/Site Address Change' )` *ES Score: 8*

File: client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	}
+
+	getContactInformationNavItem() {
+		return (
+			<VerticalNavItemMulti
+				path={ domainAddNew( this.props.selectedSite.slug ) }
+				onClick={ this.handlePickCustomDomainClick }
+				materialIcon="chrome_reader_mode"
+				text={ this.props.translate( 'Update your contact information' ) }
+				description={ this.props.translate( 'Privacy protection: on' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Privacy protection' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-domain-management-status-screen)

File: client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		);
+	}
+
+	getTransferYourDomainNavItem() {
+		return (
+			<VerticalNavItemMulti
+				path={ domainAddNew( this.props.selectedSite.slug ) }
+				onClick={ this.handlePickCustomDomainClick }
+				materialIcon="cached"
+				text={ this.props.translate( 'Transfer your domain' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Adding your domain transfer' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-domain-management-status-screen)

File: client/my-sites/domains/domain-management/edit/domain-types/wpcom-domain-type.jsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+						: `https://${ this.props.domain.name }/wp-admin/index.php?page=my-blogs#blog_row_${ this.props.selectedSite.ID }`
+				}
+				external={ ! isWpcomDomain }
+				onClick={
+					isWpcomDomain ? this.handleChangeSiteAddressClick : this.handleEditSiteAddressClick
+				}
+				materialIcon="language"
+				text={
+					isWpcomDomain
+						? this.props.translate( 'Change WordPress.com Site Address' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/wpcom-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **26** times:
`translate( 'WordPress.com Blog/Site Address Change' )` *ES Score: 8*

File: client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+		);
+	}
+
+	resolveStatus() {
+		const { domain, translate } = this.props;
+		const { registrationDate } = domain;
+
+		if ( domain.isPendingIcannVerification && domain.currentUserCanManage ) {
+			return {
+				statusText: translate( 'Action required' ),
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **46** times:
`translate( 'Action required!' )` *ES Score: 10*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/new-domain-management-status-screen)

File: client/my-sites/domains/domain-management/edit/style.scss [hunk 6]
```diff
@@ -38,3 +38,55 @@
...
+		@include domain_status_color( #C9356E );
+	}
+
+	.domain-status__card {
+		h2 {
+			font-size: 32px;
+		}
+
+		> .domain-status__icon {
+			font-size: 16px;
```
Comment (fditrapani) [inline client/my-sites/domains/domain-management/edit/style.scss]: I know I have the size set to 16px in my design but we should remove this so it defaults to the body font size which I think is 15px. There was a proposal to update our type sizes that was implemented in our Figma libraries but never in code.

File: client/my-sites/domains/domain-management/edit/style.scss [hunk 7]
```diff
@@ -38,3 +38,55 @@
...
+		color: $color;
+		> svg {
+			fill: $color;
+		}
+	}
+}
+
+.domain-types__container {
+	.status-success {
+		@include domain_status_color( #008a20 );
```
Comment (fditrapani) [inline client/my-sites/domains/domain-management/edit/style.scss]: We should be using CSS variables for these colours. They are all defined here: 

https://github.com/Automattic/wp-calypso/blob/master/packages/calypso-color-schemes/src/shared/color-schemes/_default.scss

and are implemented like this:
`@include domain_status_color(  var( --color-success-50 ) );`

**Comment by sgomes** [comment]
This PR added a `moment` dependency to a file that's largely used all across Calypso. This caused e.g. the `/log-in` entry point to go up by a significant amount of bytes, even though the new function is only used in one file, in a very isolated part of Calypso.

To avoid situations like this, it's preferable to maintain a directory of utilities (with an index file and a `package.json` file to enable tree-shaking, if need be), rather than place many unrelated utilities in a single file. I'll prepare a PR modularising `lib/domain` utilities to get that going.

**Comment by akirk** [comment]
@hambai Could you please add a screenshot? This would be helpful to our translators. Thanks!

**Pull request merged**